pruntime/enclave/src/contracts/balance.rs

```diff
@@ -56,8 +59,9 @@ pub enum Request {
 pub struct TransferData {
     dest: AccountIdWrapper,
     amount: chain::Balance,
-    signature: Option<Vec<u8>>,
+    signature: Vec<u8>,
     sequence: SequenceType,
+    machine_id: [u8; 16],
```

Comment:
`machine_id` is unnecessary. The miner who submits the transaction back to the blockchain is associated with the `machin_id` if the miner is registered. The pallet can read the machine_id and the public key on-chain with the miner's address.